inc/Mailer.class.php

```diff
@@ -198,15 +199,21 @@ public function setBody($text, $textrep = null, $htmlrep = null, $html = null, $
         if(is_null($html)) {
             $html = $text;
             $html = hsc($html);
-            $html = preg_replace('/^-----*$/m', '<hr >', $html);
+            $html = preg_replace('/^----+$/m', '<hr >', $html);
             $html = nl2br($html);
         }
         if($wrap) {
             $wrap = rawLocale('mailwrap', 'html');
             $html = preg_replace('/\n-- <br \/>.*$/s', '', $html); //strip signature
+            $html = str_replace('@EMAILSIGNATURE@', '', $html); //strip @EMAILSIGNATURE@
             $html = str_replace('@HTMLBODY@', $html, $wrap);
         }
 
+        if(strpos($text, '@EMAILSIGNATURE@') === false) {
+            $text = rtrim($text);
+            $text .= "\n\n\n\n-- \n" . $lang['email_signature'] . ":\n@DOKUWIKIURL@\n";
```

Comment:
This line is confusing.  I would think .= '@EMAILSIGNATURE@'; (best as it keeps the text signature replacement code in one place further down) or use DOKU_URL rather than @DOKUWIKIURL@.